### PR TITLE
When creating a version, test for access against the versionable FedoraResource

### PR DESCRIPTION
**Pull request by whikloj**
Addresses [FCREPO-1830](https://jira.duraspace.org/browse/FCREPO-1830)

### CODE REVIEW

File: src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 1]
```diff
@@ -125,12 +129,42 @@ private FedoraResource locateResource(final Path path, final Session session) {
...
+     * Get the versionable FedoraResource for this version resource
+     *
+     * @param resource the Version resource
+     * @return the base versionable resource or the version if not found.
+     */
+    private FedoraResource getBaseVersion(final FedoraResource resource) {
+        final Session internalSession = sessionFactory.getInternalSession();
+        try {
+            final VersionHistory base = ((Version) resource.getNode()).getContainingHistory();
+            if (base.hasProperty("jcr:versionableUuid")) {
```
Comment (awoods) [inline src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: Please pull the "jcr:versionableUuid" string into a `private static final` data member.

File: src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 2]
```diff
@@ -125,12 +129,42 @@ private FedoraResource locateResource(final Path path, final Session session) {
...
+                LOGGER.debug("versionableUuid : {}", base.getProperty("jcr:versionableUuid").getValue().getString());
```
Comment (awoods) [inline src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: You could probably create a local variable that holds: `base.getProperty(JCR_VERSIONABLE_UUID).getValue().getString())`

File: src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java (hunk 1 above)
Comment (ajs6f) [inline src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: It might be available in a MODE constants class. Would be worth doing an IDE search.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo-module-auth-webac/commit/ad8d037dea40feea7cb661bc5e539cbc79a54c6c

**Pull request closed without merging**